src/App.tsx

```diff
@@ -1,24 +1,243 @@
+import {
+  getTodos,
+  addTodo,
+  removeTodo,
+  toggleTodo,
+  changeTodoTitle,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Filter, ErrorType } from './utils/enums';
+import { TodoList } from './components/TodoList';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+const USER_ID = 10380;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [filterStatus, setFilterStatus] = useState<Filter>(Filter.All);
+  const [query, setQuery] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [processedTodos, setProcessedTodos] = useState<number[]>([]);
+
+  function filterTodos() {
+    switch (filterStatus) {
+      case Filter.All:
+        return todos;
+      case Filter.Completed:
+        return todos.filter(todo => todo.completed);
+      case Filter.Active:
+        return todos.filter(todo => !todo.completed);
+      default:
+        return todos;
+    }
+  }
+
+  const visibleTodos = useMemo(filterTodos, [todos, filterStatus]);
+  const activeTodos = todos?.filter(todo => !todo.completed);
+  const isCompletedTodos = todos.some(todo => todo.completed);
+  const allCompletedTodosId = todos
+    .filter(todo => todo.completed)
+    .map(todo => todo.id);
+
+  const getDataFromServer = useCallback(() => {
+    getTodos(USER_ID)
+      .then(data => setTodos(data))
+      // eslint-disable-next-line no-console
+      .catch(() => setError(ErrorType.Get));
+  }, []);
+
+  const addDataToServer = (title: string) => {
+    setIsLoading(true);
+    addTodo(USER_ID, title)
+      .then(data => {
+        setIsLoading(false);
+        setTodos([...todos, data]);
+        setTempTodo(null);
+      })
+      .catch(() => setError(ErrorType.Post));
+  };
+
+  const removeDataFromServer = (todoId: number) => {
+    removeTodo(todoId)
+      .then(() => {
+        setTodos(currTodos => currTodos.filter(todo => todo.id !== todoId));
+        setProcessedTodos(currState => currState.filter(id => id !== todoId));
+      })
+      .catch(() => setError(ErrorType.Delete));
+  };
+
+  const toggleDataOnServer = (todoId: number, completed: boolean) => {
+    toggleTodo(todoId, completed)
+      .then(() => {
+        setTodos(currTodos => currTodos.map(todo => {
+          if (todo.id !== todoId) {
+            return todo;
+          }
+
+          return {
+            ...todo,
+            completed,
+          };
+        }));
+        setProcessedTodos(currState => currState.filter(id => id !== todoId));
+      })
+      .catch(() => setError(ErrorType.Patch));
+  };
+
+  const changeDataonServer = (todoId: number, title: string) => {
```

Comment:
```suggestion
  const changeTodoTitle = (todoId: number, title: string) => {
```
as for me `data` is not clear enough
the same for `toggleTodoStatus` - but not critical - up to you